### PR TITLE
feat(planner): resolve schema in plan node

### LINKED ISSUE #245: proposal(catalog): move name field to `ColumnDesc` definition instead of `ColumnCatalog`
```rust
pub struct ColumnDesc {
    datatype: DataType,
    is_primary: bool,
}
```
and 
```rust
pub struct ColumnCatalog {
    id: ColumnId,
    name: String,
    desc: ColumnDesc,
}
```
should it be like this? we can move the column name to the `ColumnDesc`.
that makes sense, and we can use ColumnDesc in the schema of `PlanNode` and `Executor`, so that we can output the column name in the query result. #231 
```rust
pub struct ColumnDesc {
    datatype: DataType,
    name: String,
    is_primary: bool,
}
pub struct ColumnCatalog {
    id: ColumnId,
    desc: ColumnDesc,
}
```

### PR DESCRIPTION
**Pull request by pleiadesian**
Close #245 

Signed-off-by: Zulai Wang <zl31wang@gmail.com>

### CODE REVIEW

File: src/optimizer/plan_nodes/logical_copy_from_file.rs [hunk 1]
```diff
@@ -18,13 +18,21 @@ pub struct LogicalCopyFromFile {
     format: FileFormat,
     /// The column types.
     column_types: Vec<DataType>,
```
Comment (wangrunji0408) [inline src/optimizer/plan_nodes/logical_copy_from_file.rs:20]: Now this field is redundant.
```suggestion
```

File: src/optimizer/plan_nodes/logical_values.rs [hunk 2]
```diff
@@ -13,13 +13,19 @@ use crate::types::DataType;
 #[derive(Debug, Clone, Serialize)]
 pub struct LogicalValues {
     column_types: Vec<DataType>,
```
Comment (wangrunji0408) [inline src/optimizer/plan_nodes/logical_values.rs:15]: Now this field is redundant.
```suggestion
```

File: src/optimizer/plan_nodes/physical_simple_agg.rs [hunk 3]
```diff
@@ -44,8 +35,22 @@ impl PlanTreeNodeUnary for PhysicalSimpleAgg {
...
+            .map(|agg_call| {
+                use crate::binder::AggKind::*;
+                let name = match agg_call.kind {
+                    Avg => "avg",
+                    RowCount | Count => "count",
+                    Max => "max",
+                    Min => "min",
+                    Sum => "sum",
+                }
+                .to_string();
```
Comment (wangrunji0408) [inline src/optimizer/plan_nodes/physical_simple_agg.rs:50]: What about:
```suggestion
                let name = format!("{:?}", agg_call.kind);
```

File: src/optimizer/plan_nodes/logical_projection.rs [hunk 4]
```diff
@@ -50,10 +50,25 @@ impl PlanTreeNodeUnary for LogicalProjection {
...
-    fn out_types(&self) -> Vec<DataType> {
+    fn schema(&self) -> Vec<ColumnDesc> {
+        let child_schema = self.child.schema();
         self.project_expressions
             .iter()
-            .map(|expr| expr.return_type().unwrap())
+            .map(|expr| {
+                let name = match expr {
+                    BoundExpr::ColumnRef(column_ref) => column_ref.desc.name().to_string(),
+                    BoundExpr::TypeCast(type_cast) => match expr {
```
Comment (wangrunji0408) [inline src/optimizer/plan_nodes/logical_projection.rs:60]:
```suggestion
                    BoundExpr::TypeCast(type_cast) => match type_cast.expr {
```

**Review by wangrunji0408 — COMMENTED** [review]
LGTM!

File: src/optimizer/plan_nodes/logical_copy_from_file.rs (hunk 1 above)
Comment (pleiadesian) [reply src/optimizer/plan_nodes/logical_copy_from_file.rs:20]: It is used by `column_types()`, which returns the reference to `column_types` to caller. So, I think redundantly maintaining a `column_types` field is acceptable.

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**